### PR TITLE
[msal-node] Cache Lookup - 1: Logical keys for cache entities

### PR DESCRIPTION
**Pull request by sameerag**
This PR generates `AccountId` and `CredentialId` as a part of the `CredentialEntity` and `AccountEntity` which will be later used in looking up cache elements as needed.

#### Schema
`Account` and `Credential` cache has the below schema for keys:

| Entity type | Key components                                               |
| ----------- | ------------------------------------------------------------ |
| Account     | <home_account_id>-< environment>-<realm*>                     |
| Credential  | <home_account_id*>-< environment>-<credential_type>-<client_id>-<realm*>-<target*> |

But to tie up a `Credential` with `Account`, the keys are logically divided as follows:

#### Logical key components
| Field         | Format                                | Applicable to          |
| ------------- | ------------------------------------- | ---------------------- |
| account_id    | <home_account_id>-< environment>       | Account <br>Credential |
| realm             | < realm>                               | Account                |
| credential_id | <credential_type>-<client_id>-< realm> | Credential             |
| target        | < target>                              | Credential             |

### CODE REVIEW

File: lib/msal-common/src/unifiedCache/entities/AccountEntity.ts [hunk 1]
```diff
@@ -25,17 +25,31 @@ export class AccountEntity {
...
+    generateAccountId(): string {
+        const accountId: Array<string> = [this.homeAccountId, this.environment];
+        return accountId.join(Separators.CACHE_KEY_SEPARATOR).toLowerCase();
+    }
+
+    /**
+     * Generate Account Id key component as per the schema: <home_account_id>-<environment>
+     */
+    generateRealm(): string {
+        return (this.realm || "").toLowerCase();
```
Comment (sangonzal) [inline lib/msal-common/src/unifiedCache/entities/AccountEntity.ts:40]: I think empty space is used in multiple places, consider adding it to the constants file

Comment (pkanher617) [reply lib/msal-common/src/unifiedCache/entities/AccountEntity.ts:40]: +1 for EMPTY_STRING constant

File: lib/msal-common/src/unifiedCache/entities/Credential.ts [hunk 2]
```diff
@@ -3,13 +3,62 @@
...
     secret: string;
-};
+    familyId?: string;
+    realm?: string;
+    target?: string;
+
+    /**
+     * Generate Account Id key component as per the schema: <home_account_id>-<environment>
+     */
+    generateAccountId(): string {
```
Comment (pkanher617) [inline lib/msal-common/src/unifiedCache/entities/Credential.ts:24]: Can these be used as utils for cache lookup (i.e. generate keys for looking up a specific cred)?

Comment (sameerag) [reply lib/msal-common/src/unifiedCache/entities/Credential.ts:24]: So generating accountId for an account object instance should be separate from generating it from a set of values. I will try coming up with common code as needed, for now I added static helpers in `CacheHelper.ts` in a later PR to address this. Point noted though.

**Pull request merged**